009-ui-and-admin-server-security.md

```diff
@@ -0,0 +1,84 @@
+Once the userId and password have been entered, the UI validates the credentials on Zookeeper using `passport.js`.
+The result that is returned will be marked as valid or invalid.
+If the credentials are valid, they are stored in the session data associated with that browser session.
+When the UI requires data from the Admin Server, the credentials are recovered from the session data and placed as an auth header on the request.
+The Admin Server receives the request and configures the Kafka client in preparation for calling the Kafka broker.
+The SCRAM credentials are retrieved from the auth header and are added to the Kafka client configuration properties before executing the call to the Kafka broker.
+From thereon, the Kafka authentication and authorization mechanism works in an identical way to the description above for the Kafka Clients.
+
+This mechanism should be straightforward to setup using a `passport.js` strategy.
+The Strimzi UI is served from an Express server and that Express server will be inside the Strimzi cluster allowing the Express Server to talk to the Zookeeper on the secure port 2181.
+The login would not go through the Admin Server which hides the process from the external network but it may be desirable to add a `login` endpoint to the Admin Server which would allow session based applications to use the Admin Server in a natural way.
+
+The Strimzi Admin Server API could be called directly by the user creating an auth header on the request.
+Like the UI flow, the credentials would be retrieved from the auth header and placed into the Kafka client properties as SCRAM credentials.
+
+## OAuth 2
+
+![Kafka Client using Oauth2](./images/oauth-client.png)
+
+In the OAuth 2 scenario, Strimzi currently requires the user to configure the Kafka client with a clientId and secret.
+The client will then obtain an access token from the identity provider which is KeyCloak and that token flows to the broker.
+In the broker, calls to KeyCloak will be made based on the type of token that was requested by the client and passed to the broker.
+
+![Strimzi UI using Oauth2](./images/oauth-ui.png)
+
+In the UI model, the user will be presented with a login dialogue box when the UI is loaded into the browser and requires the user to enter a userId and password.
+The userId needs to be pre-defined to KeyCloak through the KeyCloak user interface.
+When the user enters their userId and password, the UI will contact KeyCloak to authorise the user and obtain an access token.
+This process will be controlled by `passport.js`.
+KeyCloak will either return an access token or reject the request.
+If an access token is returned, it will be stored in the session data for the session.
+When the UI requires data from the Admin Server, the token is recovered from the session data and placed as an auth header on the request.
+The Admin Server receives the request and configures the Kafka client in preparation for calling the Kafka broker.
```

Comment:
Will the admin server keep some long held connections? @mstruk can provide more details. But I think there are so re-authentications happening during the lifetime of the connection. So if the API server keep sany long term connections open, it might need refresh token or something to be able to handle them. If they are short connections only ... e.g. connect, list topics, disconnect ... it should be fine I guess.

Reply:
One can always start a new connection if the existing one closes due to a failed re-authentication.